app/jobs/valkyrie_create_derivatives_job.rb

```diff
@@ -1,23 +1,24 @@
 # frozen_string_literal: true
 class ValkyrieCreateDerivativesJob < Hyrax::ApplicationJob
   queue_as Hyrax.config.ingest_queue_name
-  def perform(_file_set_id, file_id, _filepath = nil)
+  def perform(file_set_id, file_id, _filepath = nil)
     file_metadata = Hyrax.custom_queries.find_file_metadata_by(id: file_id)
     return if file_metadata.video? && !Hyrax.config.enable_ffmpeg
     # Get file into a local path.
     file = Hyrax.storage_adapter.find_by(id: file_metadata.file_identifier)
     # Call derivatives with the file_set.
     derivative_service = Hyrax::DerivativeService.for(file_metadata)
     derivative_service.create_derivatives(file.disk_path)
+    reindex_parent(file_set_id)
   end
 
   private
 
-  def query_service
-    Hyrax.query_service
-  end
-
-  def storage_adapter
-    Hyrax.storage_adapter
+  def reindex_parent(file_set_id)
+    file_set = Hyrax.query_service.find_by(id: file_set_id)
+    parent = Hyrax.custom_queries.find_parent_work(resource: file_set)
+    return unless parent.thumbnail_id == file_set.id
+    user = User.find_by_user_key(file_set.depositor)
+    Hyrax.publisher.publish('object.metadata.updated', object: parent, user: user)
```

Comment:
it doesn't seem right to publish `object.metadata.updated` here? we're not actually updating the object metadata.

if the goal is to reindex, shouldn't we just do that explicitly and avoid triggering other application behavior that might expect there to have been an actual change attributable to `user`?
 ```suggestion
    Hyrax.index_adapter.save(resource: parent)
```

Reply:
i think this also avoids the need to query for a user, who didn't really have anything to do with this change.

Reply:
I was under the impression that publishing was the preferred way in general. Thanks for clearing that up.

Reply:
i think the contract with publishing events is that Hyrax will:
  - always publish them when the event occurs;
  - never publish them otherwise.

downstream applications need to be able to rely on events to trigger at the right times and not to invoke listener behavior when it's not appropriate.